frontend/src/component/changeRequest/ChangeRequest/Changes/Change/ChangeOverwriteWarning/strategy-change-diff-calculation.ts

```diff
@@ -142,9 +142,31 @@ export function getStrategyChangesThatWouldBeOverwritten(
     change: IChangeRequestUpdateStrategy,
 ): ChangesThatWouldBeOverwritten | null {
     const fallbacks = { segments: [], variants: [], title: '' };
+
+    const withSortedSegments = (() => {
+        if (change.payload.segments && currentStrategyConfig?.segments) {
+            const changeCopy = {
+                ...change,
+                payload: {
+                    ...change.payload,
+                    segments: [...(change.payload.segments ?? [])].sort(),
+                },
+            };
+
+            const currentCopy = {
+                ...currentStrategyConfig,
+                segments: [...currentStrategyConfig.segments].sort(),
+            };
+
+            return { current: currentCopy, change: changeCopy };
+        } else {
+            return { current: currentStrategyConfig, change: change };
```

Comment:
Nitpicking, but maybe it would look cleaner if we early returned instead.
```ts
if (!change.payload.segments || !currentStrategyConfig?.segments) {
  return { current: currentStrategyConfig, change };
}
// ...
```

Reply:
Yeah, good shout.